kpops/components/base_components/base_defaults_component.py

```diff
@@ -193,31 +193,44 @@ def defaults_from_yaml(path: Path, key: str) -> dict:
 
 
 def get_defaults_file_paths(
-    config: KpopsConfig, environment: str | None
-) -> tuple[Path, Path]:
+    pipeline_path: Path, config: KpopsConfig, environment: str | None
+) -> list[Path]:
     """Return the paths to the main and the environment defaults-files.
 
     The files need not exist, this function will only check if the dir set in
     `config.defaults_path` exists and return paths to the defaults files
     calculated from it. It is up to the caller to handle any false paths.
 
+    :param pipeline_path: The path to the pipeline.yaml file
     :param config: KPOps configuration
     :param environment: Environment
     :returns: The defaults files paths
     """
-    defaults_dir = Path(config.defaults_path).resolve()
-    main_default_file_path = defaults_dir / Path(
-        config.defaults_filename_prefix
-    ).with_suffix(".yaml")
-
-    environment_default_file_path = (
-        defaults_dir
-        / Path(f"{config.defaults_filename_prefix}_{environment}").with_suffix(".yaml")
-        if environment is not None
-        else main_default_file_path
-    )
+    default_paths = []
+
+    if not pipeline_path.is_file():
+        message = f"No pipeline.yaml found in directory: {pipeline_path}"
+        raise FileNotFoundError(message)
+
+    path = pipeline_path.resolve()
+    pipeline_base_dir = config.pipeline_base_dir.resolve()
+    if pipeline_base_dir not in path.parents:
+        message = f"The given pipeline base path {pipeline_base_dir} is not part of the pipeline path {path}"
+        raise RuntimeError(message)
+    while pipeline_base_dir != path:
+        environment_default_file_path = (
+            path.parent / f"{config.defaults_filename_prefix}_{environment}.yaml"
+        )
+        if environment_default_file_path.is_file():
+            default_paths.append(environment_default_file_path)
+
+        defaults_yaml_path = path.parent / f"{config.defaults_filename_prefix}.yaml"
+        if defaults_yaml_path.is_file():
+            default_paths.append(defaults_yaml_path)
+
+        path = path.parent
 
-    return main_default_file_path, environment_default_file_path
+    return default_paths
```

Comment:
Does it make sense to extract the logic that looks for the files into a function which we could later use for the config?

Reply:
It makes sense, but let's keep everything simple and see if the core functionality of KPOps works as expected.